service/actions/channels.js

```diff
@@ -381,6 +381,11 @@ export function deleteChannel(teamId, channelId) {
             return;
         }
 
+        if (beforeDelete) {
+            const {action, args, _dispatch, _state} = beforeDelete;
```

Comment:
I like the idea of using the event emitter here, seems like it would handle this situation nicely.